### PR TITLE
[BUU] Remove Stimulus Reflex from Products screen

### LINKED ISSUE #12331: [BUU] [Infinite loader] - Remove Stimulus Reflex from Products Page
In some circumstances, the new bulk edit products screen will fail to load products, leaving an infinite loader, and a poor user wondering if anything was still happening.
- Split from #11752 because these issues have different impacts and will be solved separately

**Action Items:**
Remove SR From:
- [x] Product screen loading, filtering and saving (https://github.com/openfoodfoundation/openfoodnetwork/pull/12328)
- [x] Product delete action (https://github.com/openfoodfoundation/openfoodnetwork/issues/12398)
- [ ] #12551

## Steps to reproduce
This was last replicated on fr_prod, at approx 9:30am AEST: https://github.com/openfoodfoundation/openfoodnetwork/issues/11752#issuecomment-2024144334
Sometimes it works fine, but maybe if the server is busier, it doesn't.

Image

### PR DESCRIPTION
**Pull request by dacook**
#### What? Why?

- Closes #12331 

After much discussion, I have removed most of the StimulusReflex from the new Products screen. The plan is to make it more dynamic with Turbo/AJAX.

SR is still there for deleting records, which could still suffer the loading issues we had. (It's also used for the Edit Image modal, but this shouldn't have the loading issue). 
I expect to continue with removing those, but it would be more efficient to transfer them directly to a Turbo/AJAX implementation later.

#### What should we test?
 All the things!! 🫠 (almost)

* Page load: shows products as expected
* Filtering: by search term, products, category
* Pagination: 
   * choose number per page
   * browse pages
* Submit changes
   * PRoducts
   * Variants
   * Form errors (eg blank product name)
* New variant
   * Form errors (eg blank unit)
* Product actions work as before
   * Edit image (still uses a little bit of StimulusReflex)
   * Edit record
   * Delete (still uses StimulusReflex to load products after)
   * Clone
* Variant actions: these work the same as product, so no need to test again.

#### Release notes

Changelog Category (reviewers may add a label for the release notes):

- [ ] User facing changes
- [ ] API changes (V0, V1, DFC or Webhook)
- [ ] Technical changes only
- [x] Feature toggled

The title of the pull request will be included in the release notes.

### CODE REVIEW

File: config/routes/admin.rb [hunk 1]
```diff
@@ -71,7 +71,7 @@
 
     constraints FeatureToggleConstraint.new(:admin_style_v3) do
       resources :products, to: 'products_v3#index', only: :index do
-        patch :bulk_update, on: :collection
+        patch :bulk_update, to: 'products_v3#bulk_update', as: :bulk_update
```
Comment (rioug) [inline config/routes/admin.rb:74]: Maybe this, I think you forgot the `collection`
```suggestion
       collection do
         patch :bulk_update, to: 'products_v3#bulk_update', as: :bulk_update
       end
```

**Review by mkllnk — APPROVED** [review]
I did only a quick review but didn't find anything obviously wrong. It looks easier than I thought to remove SR from the page.

**Review by rioug — APPROVED** [review]
Looks fine !

**Comment by dacook** [comment]
Sorry @filipefurtad0 I seem to have missed something in the rebase, I'll have to fix it up for testing tomorrow instead.

**Comment by dacook** [comment]
@filipefurtad0 I've fixed the rebase, and added testing notes. Please test away! 🧪

**Comment by filipefurtad0** [comment]
Hi @dacook ,

Unfortunately, I just lost all my notes, but the summary is - let's merge.

I went through all the test you indicate (thank you :pray:) and spotted only a minor regression, when editing a product/variant, with any given filter:
- before this PR, after saving changes, the filters were kept (only a subset of variants is displayed, accordingly)
- after this PR, after saving changes, the filters were cleared (all available variants are displayed)

**Not related to this PR**
I found two usability issues as well:
- click "edit variant"; instead of editing, click "cancel" -> redirect should occur to the products page, but instead, redirects to the page of that product (perhaps per-existent, and not a priority)
- editing "Units" from a variant; delete the existing value; change your mind and wish to do something else -> you can't click away from the edit unit sub menu, neither Esc or clicking elsewhere works. That means you need to input a value. Only workaround is refreshing the page.

Also, I've merged this branch on top of the specs for BUU and spotted no additional failing specs.

Merging!

**Comment by dacook** [comment]
Hi Filipe, thanks for testing this one! 

> I went through all the test you indicate (thank you 🙏) and spotted only a minor regression, when editing a product/variant, with any given filter:
> 
> * before this PR, after saving changes, the filters were kept (only a subset of variants is displayed, accordingly)
> * after this PR, after saving changes, the filters were cleared (all available variants are displayed)

Good spotting, noted. I've fixed this in my next branch.

> **Not related to this PR** I found two usability issues as well:


I thought it would be a good opportunity to test out the general usability of the page ;)

> * click "edit variant"; instead of editing, click "cancel" -> redirect should occur to the products page, but instead, redirects to the page of that product (perhaps per-existent, and not a priority)

> * editing "Units" from a variant; delete the existing value; change your mind and wish to do something else -> you can't click away from the edit unit sub menu, neither Esc or clicking elsewhere works. That means you need to input a value. Only workaround is refreshing the page.

I can see how this is a bit annoying, but I found it usable: I was able to click "discard changes" to reset, or press CTRL/CMD-Z to undo. Or just type it in again ;)

However there is a problem with units when creating a new variant.

> Also, I've merged this branch on top of the specs for BUU and spotted no additional failing specs.

🎉

**Pull request merged**